riocli/device/report.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+
+import click
+from click_help_colors import HelpColorsCommand
+from rapyuta_io.clients import SharedURL
+
+from riocli.config import new_client
+from riocli.constants import Colors, Symbols
+from riocli.device.util import name_to_guid, report_device_api_call
+from riocli.utils.spinner import with_spinner
+
+
+@click.command(
+    "report",
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option(
+    "--force", "-f", "--silent", "force", is_flag=True, help="Skip confirmation"
+)
+@click.option(
+    "--share",
+    "-s",
+    is_flag=True,
+    help="Share the device logs with Rapyuta support",
+    default=False,
+)
+@click.option(
+    "--expiry",
+    "-e",
+    help="Expiry time for the shared URL in days",
+    default=7,
+    required=False,
+    show_default=True,
+)
+@click.argument("device-name", type=str)
+@name_to_guid
+@with_spinner(text="Upload debugging files...")
+def report_device(
+    device_name: str,
+    device_guid: str,
+    force: bool,
+    share: bool,
+    expiry: int,
+    spinner=None,
+) -> None:
+    """
+    Report a device and get its debug logs in uploads section.
+    """
```

Comment:
todo: Please add an elaborate docstring. You can refer this file to get an idea
https://github.com/rapyuta-robotics/rapyuta-io-cli/blob/devel/riocli/apply/__init__.py